page_separator.py

```diff
@@ -29,33 +30,36 @@ def page_type(page: PageObject) -> PageType:
 def insert_suffix(pdf: str, suffix: str) -> str:
     return pdf.split(".pdf")[0] + suffix + ".pdf"
     
-
-if __name__ == '__main__':
-    parser = argparse.ArgumentParser(description="Splits a PDF into two PDFs by letter and legal page size.")
-    parser.add_argument("source_pdf", help="Name of the source PDF with mixed page size, E.g. SamplePrint-MultipleSize.pdf")
-    parser.add_argument("--letter_pdf", help="Name of the PDF to write to containing letter pages (default: <source_pdf>_letter.pdf)")
-    parser.add_argument("--legal_pdf", help="Name of the PDF to write to containing legal pages (default: <source_pdf>_legal.pdf)")
-    args = parser.parse_args()
-    
-    letter_pdf = args.letter_pdf if args.letter_pdf else insert_suffix(args.source_pdf, "_letter")
-    legal_pdf = args.legal_pdf if args.legal_pdf else insert_suffix(args.source_pdf, "_legal")
-    
-    reader = PdfFileReader(open(args.source_pdf, "rb"))
-    letter_writer = PdfFileWriter()
+def main(source_pdf: str, legal_pdf: str, letter_pdf: str):
+    reader = PdfFileReader(open(source_pdf, "rb"))
     legal_writer = PdfFileWriter()
+    letter_writer = PdfFileWriter()
     
     for i in range(reader.getNumPages()):
         page = reader.getPage(i)
         ptype = page_type(page)
-        if ptype == PageType.LETTER:
-            letter_writer.addPage(page)
-        elif ptype == PageType.LEGAL:
+        if ptype == PageType.LEGAL:
             legal_writer.addPage(page)
+        elif ptype == PageType.LETTER:
+            letter_writer.addPage(page)
         else:
             sys.exit("Could not infer page size for page #{}".format(i))

```

Comment:
hmm. not entirely sure how helpful that is

Reply:
From the way I see it, the end user are presented with the troublesome pages instead of having to go look them up themselves. If it is only one page for example, they can choose to print the legal and letter pdfs, and judge what to do with the unidentified. It's not like they can't run the separator again and overwrite them if it's not what they want. As long as, the message is printed out that there are unidentified pages, I think it should be fine.